### PR TITLE
Home page takes users and posts (only social so far) from database

### PR DESCRIPTION
**Pull request by vaal1212**
Home page now works with database (all REST)

### CODE REVIEW

**Comment by DanaMartynovska** [comment]
+

**Comment by DanaMartynovska** [comment]
Merge should be considered by the team, cause of the project design structure.

File: crossfit/settings.py [hunk 1]
```diff
@@ -143,6 +141,7 @@
 
 # create root dir for all images
 MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
+print ("MEDIA ROOT:", MEDIA_ROOT)
```
Comment (StanislavDmt13) [inline crossfit/settings.py:144]: Remove prints from your code

File: frontend/comments_views.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+class CreateListTrainingPostCommentsView(mixins.ListModelMixin, mixins.CreateModelMixin, GenericAPIView):
+    serializer_class = CommentSerializer
+
+    def get_queryset(self):
+        print(self.request.GET['post_type'])
+        post_id = int(self.kwargs['post_id'])
+        post = None
+        if self.request.GET['post_type'] == 'social':
+            post = SocialPost.objects.filter(id = post_id)[0]
```
Comment (StanislavDmt13) [inline frontend/comments_views.py:22]: .filter(id = post_id).first()
If SocialPost with post_id is undefined you get 500n error

File: frontend/comments_views.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        post_comments = post.comments.all().prefetch_related('author')
+        return post_comments
+    #END def get_queryset(self):
+
+    def get(self, request, *args, **kwargs):
+        return self.list(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        print(request.POST['post_type'])
+        new_comment = Comment(
```
Comment (StanislavDmt13) [inline frontend/comments_views.py:34]: If you want to create new object you need to do it with serializer for validate your input data

File: frontend/homePage_views.py [hunk 4]
```diff
@@ -46,4 +53,68 @@ def login_user(request):
             messages.success(request, "Username or password is not correct.")
     context ={}
     context["no_login_link"] = True
-    return render(request, 'login_user.html', context)
\ No newline at end of file
+    return render(request, 'login_user.html', context)
+
+
+
+def getRandomObjects(modelManager, number_needed):
```
Comment (StanislavDmt13) [inline frontend/homePage_views.py:60]: For getting random object you can use random sorting. See resource https://stackoverflow.com/questions/22816704/django-get-a-random-object

File: frontend/urls.py [hunk 5]
```diff
@@ -5,13 +5,22 @@
...
 
 user_url_patterns = [
     path('', userPage_views.user_profile, name='user_profile'),
     path('edit/', userPage_views.edit, name="edit"),
     path('friends_list/', userPage_views.friends_list, name="friends_list"),
 ]
 
+post_url_patterns = [
+    #TODO: consider if URLS below belong to separate post_urls.py or move all URLs to one file
+    path('createsocialpost/', post_views.create_social_post, name="create_social_post"),
```
Comment (StanislavDmt13) [inline frontend/urls.py:18]: Better to create urls by this way
POST /socialposts for create new obj
GET /socialposts for get list of objs
GET /socialposts/<id:int> for get some obj
POST /socialposts/<id:int> for change some obj 
etc.

**Review by StanislavDmt13 — COMMENTED** [review]
Media files you don't need to save on github. You need to add this directory to the .gitignore.
Media files it is files what were uploaded by  your customers

File: frontend/comments_views.py (hunk 2 above)
Comment (vaal1212) [reply frontend/comments_views.py]: Yes, I knew it was not a good practice to do so, but was lazy to look for a proper way.
Resolved in "First object by list indes [] notation replaced with calls to .first() method of queryset." commit.

File: crossfit/settings.py (hunk 1 above)
Comment (vaal1212) [reply crossfit/settings.py:144]: Done. Cleaned in "debug print messages removed" commit.

File: frontend/comments_views.py (hunk 3 above)
Comment (vaal1212) [reply frontend/comments_views.py]: Found a bad solution, but OK for education:

 #This is not ideal, but the problem is:
        # I either can have a serializer, which validates external object (author), but will not serizliae nested objectes - that is will not provide authors, only IDs -CommentSerializer 
        # Output of validating serializer (ValidatingCommentSerializer):
                # {
                #     "id":104,
                #     "date_created":"2022-09-04T16:11:27.499011+03:00",
                #     "author":1,
                #     "comment_text":"2nd comment"
                # }

        # or I can have serializer, which will provide nested author objects in JSON, but will not validate nested relation - CommentSerializer
        #Output of non validating serializer (CommentSerializer). Note full author information pulled automatically (as I need it for frontend):
                # {
                #     "id":104,
                #     "date_created":"2022-09-04T16:11:27.499011+03:00",
                #     "author":"OrderedDict("[
                #         "(""id",
                #         1),
                #         "(""photo",
                #         "/media/media/kitty_.jpg"")",
                #         "(""username",
                #         "admin"")",
                #         "(""birthday",
                #         "None)",
                #         "(""email",
                #         "admin@local.host"")",
                #         "(""phone",
                #         "None)",
                #         "(""short_bio",
                #         "sadfasd asdfasd\r\n\r\nasdfasdfsad"")"
                #     ]")",
                #     "comment_text":"2nd comment"
                # }
        #So I made both just to see how to make validation work.

File: frontend/homePage_views.py (hunk 4 above)
Comment (vaal1212) [reply frontend/homePage_views.py]: Have see this option, but it is reportedly inefficient (depending on backend), so I think better to have 'manual' solution.
    https://code.djangoproject.com/ticket/5267

File: frontend/urls.py (hunk 5 above)
Comment (vaal1212) [reply frontend/urls.py:18]: Corrected in "Post creation and view URLs corrected" commit.

File: frontend/comments_views.py (hunk 3 above)
Comment (vaal1212) [reply frontend/comments_views.py]: The issue is resolved with argument to required=False in nested serializer field. Still not super clean - required addition of the author to the newly created post object, but better than separate serializer.

**Pull request merged**